daemon/endpoint.go

```diff
@@ -246,6 +249,47 @@ func (d *Daemon) createEndpoint(ctx context.Context, epTemplate *models.Endpoint
 		}
 	}
 
+	// Static pods (mirror pods) might be configured before the apiserver
+	// is available or has received the notification that includes the
+	// static pod's labels. In this case, start a controller to attempt to
+	// resolve the labels.
+	if ep.K8sNamespaceAndPodNameIsSet() && k8s.IsEnabled() {
+		k8sLabelsConfigured := len(addLabels) != 0
+		if k8sLabelsConfigured {
+			// If there are labels, but no pod namespace then it's
+			// likely that there are no k8s labels at all. Resolve.
+			_, k8sNamespaceSet := addLabels[k8sConst.PodNamespaceLabel]
+			k8sLabelsConfigured = k8sNamespaceSet
+		}
+
+		if !k8sLabelsConfigured {
+			done := make(chan struct{})
+
+			controllerName := fmt.Sprintf("resolve-labels-%d", ep.GetID())
```

Comment:
Will the endpoint ID be non-zero here? The call to `AddEndpoint` on the endpointmanager is performed immediately after this logic, and the endpointmanager is the entity which assigns an ID to an endpoint.

Reply:
Argh, good point. I can construct it based on the pod namespace/name instead.